src/weights.jl

```diff
@@ -53,9 +57,34 @@ Base.getindex(wv::W, ::Colon) where {W <: AbstractWeights} = W(copy(wv.values),
     isfinite(sum) || throw(ArgumentError("weights cannot contain Inf or NaN values"))
     wv.values[i] = v
     wv.sum = sum
+    wv.positive = missing
     v
 end
 
+function Base.all(f::Base.Fix2{typeof(>=)}, wv::AbstractWeights)
```

Comment:
This may be overkill, but unfortunately that's the standard way of checking whether all entries in a vector in Julia, so that's the only solution if we want external code to be able to use this feature, without exporting a new `ispositive` function. One advantage of defining this is that if we rework the API to take a `weights` keyword argument, sampling functions will be able to allow any `AbstractArray` and code will automatically work.

Reply:
Hmm. It's easy to not hit this function, e.g., when using `all(x -> x >= 0, wv)` etc. And, of course, it covers also things like `all(>=(2), wv)`. One might also want to check non-negativity by e.g. `!any(<(0), wv)`.

So I think a dedicated separate function would be cleaner and less ambiguous. If one wants to support `AbstractArray`s one could also at some point just define a fallback
```julia
isnonneg(x::AbstractArray{<:Real})  = all(>=(0), x)
```

Reply:
What annoys me is that there's no reason why this very basic function would live in and be exported by StatsBase. And anyway if users are not aware of the fast path (be it `all(>=(0), x)` or `nonneg(x)`), they won't use it and get the slow one, so I'm not sure choosing one syntax or the other makes a difference.

We could keep this internal for now -- though defining an internal function wouldn't be better than `all(>=(0), wv)` and `!any(<(0), wv)` as we would be sure users wouldn't be able to use it. ;-)

Reply:
Yes, I would suggest only defining an internal function - that seems sufficient as it's only used in the argument checks internally. Something like `isnnonneg` seems to focus on what we actually want to check whereas defining `all` or `any` catches also other, in principle undesired, cases and hence requires a more complicated implementation with multiple checks.